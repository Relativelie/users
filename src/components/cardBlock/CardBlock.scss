@import "../../variables";

.userInfo__form {
    display: flex;
    flex-direction: column;
    gap: 10px;
    width: 438px;
    padding: 20px 12px;
    border: 1px solid #F3F3F3;
    border-radius: 8px;
}

.userInfo__header {
    @include header();
}

