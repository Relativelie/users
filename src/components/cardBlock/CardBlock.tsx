import { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useActions } from '../../hooks/useActions';
import { InputContainer } from './InputContainer/InputContainer';
import { SendForm } from './sendForm/SendForm';
import { useTypedSelector } from '../../hooks/useTypedSelector';

import './CardBlock.scss';

export const CardBlock = () => {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const { userId } = useParams();
    const [editBtnIsPressed, setEditBtnIsPressed] = useState('');
    const [inputValueChange, setInputValueChange] = useState(false);
    const {
        isDisabledForm,
    } = useTypedSelector((cardBlockState) => cardBlockState.cardBlockReducer);
    const {
        turnOffEditMode,
        turnOnEditMode,
        changeBtnAvailability,
        showRequestResult,
    } = useActions();

    useEffect(() => {
        checkForEmptiness();
    }, [inputValueChange]);

    const checkForEmptiness = () => {
        const errorClass = '.inputFieldContainer__input_warning-for-emptiness';
        const errorField = document.querySelector(errorClass);
        if (errorField !== null) changeBtnAvailability(true);
        else changeBtnAvailability(false);
    };

    const catchInputValueChange = () => {
        setInputValueChange(!inputValueChange);
    };

    const editMode = () => {
        if (isDisabledForm) {
            turnOnEditMode();
            setEditBtnIsPressed('cardBlock_editBtn_isActive');
        } else {
            turnOffEditMode();
            setEditBtnIsPressed('');
            showRequestResult(false);
        }
        checkForEmptiness();
    };

    return (
        <section className="cardBlock">
            <div className="cardBlock__container">
                <h3 className="cardBlock__header">Профиль пользователя</h3>
                <button
                    className={`cardBlock__editBtn ${editBtnIsPressed}`.trim()}
                    onClick={() => editMode()}
                    type="button"
                >
                    Редактировать
                </button>
            </div>
            <form className="cardBlock__form" action="some url">
                <InputContainer catchInputValueChange={catchInputValueChange} />
                <SendForm />
            </form>
        </section>
    );
};
