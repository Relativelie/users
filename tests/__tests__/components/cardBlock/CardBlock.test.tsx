import { screen } from '@testing-library/dom';
import { MemoryRouter as Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { CardBlock } from '../../../../src/components/cardBlock/CardBlock';
import { render } from '../../../testsSetup/test-utils';

describe('Card block component', () => {
    test('snapshot - card block component', () => {
        const tree = render(
            <Router>
                <CardBlock />
            </Router>,
        );
        expect(tree).toMatchSnapshot();
    });

    test('click to edit button', () => {
        render(
            <Router>
                <CardBlock />
            </Router>,
        );
        const editButton = screen.getByText(/редактировать/i);
        expect(editButton).toHaveClass('cardBlock__editBtn');
        userEvent.click(editButton);
        expect(editButton).toHaveClass('cardBlock__editBtn cardBlock_editBtn_isActive');
        userEvent.click(editButton);
        expect(editButton).toHaveClass('cardBlock__editBtn');
    });
});
