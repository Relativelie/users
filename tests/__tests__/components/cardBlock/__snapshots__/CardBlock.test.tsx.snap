// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card block component snapshot - card block component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="cardBlock"
      >
        <div
          class="cardBlock__container"
        >
          <h3
            class="cardBlock__header"
          >
            Профиль пользователя
          </h3>
          <button
            class="cardBlock__editBtn"
            type="button"
          >
            Редактировать
          </button>
        </div>
        <form
          action="some url"
          class="cardBlock__form"
        >
          <div
            class="cardBlock__inputContainer"
          >
            <div
              class="inputFieldContainer"
            >
              <label
                class="inputFieldContainer__label"
                for="Name"
              >
                Name
              </label>
              <input
                class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
                data-inputname="name"
                disabled=""
                placeholder="Name"
                type="text"
              />
            </div>
            <div
              class="inputFieldContainer"
            >
              <label
                class="inputFieldContainer__label"
                for="User name"
              >
                User name
              </label>
              <input
                class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
                data-inputname="user name"
                disabled=""
                placeholder="User name"
                type="text"
              />
            </div>
            <div
              class="inputFieldContainer"
            >
              <label
                class="inputFieldContainer__label"
                for="E-mail"
              >
                E-mail
              </label>
              <input
                class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
                data-inputname="e-mail"
                disabled=""
                placeholder="E-mail"
                type="email"
              />
            </div>
            <div
              class="inputFieldContainer"
            >
              <label
                class="inputFieldContainer__label"
                for="Street"
              >
                Street
              </label>
              <input
                class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
                data-inputname="street"
                disabled=""
                placeholder="Street"
                type="text"
              />
            </div>
            <div
              class="inputFieldContainer"
            >
              <label
                class="inputFieldContainer__label"
                for="City"
              >
                City
              </label>
              <input
                class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
                data-inputname="city"
                disabled=""
                placeholder="City"
                type="text"
              />
            </div>
            <div
              class="inputFieldContainer"
            >
              <label
                class="inputFieldContainer__label"
                for="Phone"
              >
                Phone
              </label>
              <input
                class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
                data-inputname="phone"
                disabled=""
                placeholder="Phone"
                type="tel"
              />
            </div>
            <div
              class="inputFieldContainer"
            >
              <label
                class="inputFieldContainer__label"
                for="Website"
              >
                Website
              </label>
              <input
                class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
                data-inputname="website"
                disabled=""
                placeholder="Website"
                type="url"
              />
            </div>
            <div
              class="inputFieldContainer"
            >
              <label
                class="inputFieldContainer__label"
                for="Comment"
              >
                Comment
              </label>
              <input
                class="inputFieldContainer__input inputFieldContainer__input_comment"
                data-inputname="comment"
                disabled=""
                placeholder=""
                type="text"
              />
            </div>
          </div>
          <div
            class="sendForm"
          >
            <div
              class="sendForm__container"
            >
              <div
                class="loading"
              />
              <button
                class="sendForm__sendBtn"
                disabled=""
                type="button"
              >
                Отправить
              </button>
            </div>
            <p
              class="sendForm__resultText"
            />
          </div>
        </form>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="cardBlock"
    >
      <div
        class="cardBlock__container"
      >
        <h3
          class="cardBlock__header"
        >
          Профиль пользователя
        </h3>
        <button
          class="cardBlock__editBtn"
          type="button"
        >
          Редактировать
        </button>
      </div>
      <form
        action="some url"
        class="cardBlock__form"
      >
        <div
          class="cardBlock__inputContainer"
        >
          <div
            class="inputFieldContainer"
          >
            <label
              class="inputFieldContainer__label"
              for="Name"
            >
              Name
            </label>
            <input
              class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
              data-inputname="name"
              disabled=""
              placeholder="Name"
              type="text"
            />
          </div>
          <div
            class="inputFieldContainer"
          >
            <label
              class="inputFieldContainer__label"
              for="User name"
            >
              User name
            </label>
            <input
              class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
              data-inputname="user name"
              disabled=""
              placeholder="User name"
              type="text"
            />
          </div>
          <div
            class="inputFieldContainer"
          >
            <label
              class="inputFieldContainer__label"
              for="E-mail"
            >
              E-mail
            </label>
            <input
              class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
              data-inputname="e-mail"
              disabled=""
              placeholder="E-mail"
              type="email"
            />
          </div>
          <div
            class="inputFieldContainer"
          >
            <label
              class="inputFieldContainer__label"
              for="Street"
            >
              Street
            </label>
            <input
              class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
              data-inputname="street"
              disabled=""
              placeholder="Street"
              type="text"
            />
          </div>
          <div
            class="inputFieldContainer"
          >
            <label
              class="inputFieldContainer__label"
              for="City"
            >
              City
            </label>
            <input
              class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
              data-inputname="city"
              disabled=""
              placeholder="City"
              type="text"
            />
          </div>
          <div
            class="inputFieldContainer"
          >
            <label
              class="inputFieldContainer__label"
              for="Phone"
            >
              Phone
            </label>
            <input
              class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
              data-inputname="phone"
              disabled=""
              placeholder="Phone"
              type="tel"
            />
          </div>
          <div
            class="inputFieldContainer"
          >
            <label
              class="inputFieldContainer__label"
              for="Website"
            >
              Website
            </label>
            <input
              class="inputFieldContainer__input  inputFieldContainer__input_warning-for-emptiness"
              data-inputname="website"
              disabled=""
              placeholder="Website"
              type="url"
            />
          </div>
          <div
            class="inputFieldContainer"
          >
            <label
              class="inputFieldContainer__label"
              for="Comment"
            >
              Comment
            </label>
            <input
              class="inputFieldContainer__input inputFieldContainer__input_comment"
              data-inputname="comment"
              disabled=""
              placeholder=""
              type="text"
            />
          </div>
        </div>
        <div
          class="sendForm"
        >
          <div
            class="sendForm__container"
          >
            <div
              class="loading"
            />
            <button
              class="sendForm__sendBtn"
              disabled=""
              type="button"
            >
              Отправить
            </button>
          </div>
          <p
            class="sendForm__resultText"
          />
        </div>
      </form>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
